app/models/ability.rb

```diff
@@ -0,0 +1,22 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+
+    if user.admin?
+      can :manage, :all
+    else
+      can :manage, Category, user_id: user.id
+      can :manage, Group, user_id: user.id
+      can :manage, Entity, author_id: user.id
+      can [:create, :update, :destroy, :index], Transaction do |transaction|
+        transaction.category.user_id == user.id
+      end
+      can :read, Transaction
+
+      # Temporary for testing (using Entity.find):
+      can(:manage, Entity, id: Entity.find_by(name: 'MyString').id) unless user.admin?
+    end
```

Comment:
- [ ] I got the error message screenshot below in the process of registering a new user, 👇 this is because of this line of code that was newly added. Kindly address this for your app to be accessible. **Note: Your app worked perfectly in the previous review.**

<img width="467" alt="codereview" src="https://github.com/jkanyi-web/BudgetBuddy/assets/103386183/0db12d0f-0692-4694-8e38-9e73d08334bf">